bin/oay/src/services/webdav/webdavfs.rs

```diff
@@ -156,10 +162,23 @@ impl DavFileSystem for WebdavFs {
 
     fn rename<'a>(&'a self, from: &'a DavPath, to: &'a DavPath) -> dav_server::fs::FsFuture<()> {
         async move {
-            self.op
-                .rename(from.as_url_string().as_str(), to.as_url_string().as_str())
-                .await
-                .map_err(convert_error)
+            let from_path = from
+                .as_rel_ospath()
+                .to_str()
+                .ok_or(FsError::GeneralFailure)?;
+            let to_path = to.as_rel_ospath().to_str().ok_or(FsError::GeneralFailure)?;
+            let res = self.op.rename(from_path, to_path).await;
+            match res {
+                Ok(_) => Ok(()),
+                Err(e) => {
+                    if from.is_collection() {
```

Comment:
Please don't handling error like this.

- If the services doesn't support move a dir, opendal should return `Unsupported`.
- If the services raise error for the file exists, opendal should return `AlreadyExists`
- If the services raise error for trying to rename a folder to file, opendal should return `NotADirectory`

Maybe we can check `to` first before calling `rename`?